app/src/test/java/com/capstone/weatherapp/CityListViewModelTest.kt

```diff
@@ -0,0 +1,47 @@
+package com.capstone.weatherapp
+
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.*
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class CityListViewModelUnitTest {
+    private val repo = mockk<WeatherRepository>(relaxed = true)
+    private lateinit var viewModel: CityListViewModel
+
+    @Before
+    fun setUp() {
+        viewModel = CityListViewModelFactory(repo).create(CityListViewModel::class.java)
+    }
+
+    @Test
+    fun `viewModel successfully refreshes cityList from the repo`() = runTest {
+        Dispatchers.setMain(UnconfinedTestDispatcher(testScheduler))
```

Comment:
this line can be moved into the @Before function since it's used in both methods